habitat-lab/habitat/tasks/rearrange/rearrange_sim.py

```diff
@@ -511,37 +535,37 @@ def _set_ao_states_from_ep(self, ep_info: RearrangeEpisode) -> None:
             ao.joint_positions = ao_pose
 
     def is_point_within_bounds(self, pos):
+        # NOTE: This check is loose: really we want the island bounds, not the full navmesh
         lower_bound, upper_bound = self.pathfinder.get_bounds()
         return all(lower_bound <= pos) and all(upper_bound >= pos)
 
     def safe_snap_point(self, pos: np.ndarray) -> np.ndarray:
         """
-        snap_point can return nan which produces hard to catch errors.
+        Returns the 3D coordinates corresponding to a point belonging
+        to the biggest navmesh island in the scenee and closest to pos.
+        When that point returns NaN, computes a navigable point at increasing
+        distances to it.
         """
-        new_pos = self.pathfinder.snap_point(pos)
-        island_radius = self.pathfinder.island_radius(new_pos)
+        new_pos = self.pathfinder.snap_point(pos, self._largest_island_idx)
+
+        max_iter = 10
+        offset_distance = 1.5
+        distance_per_iter = 0.5
+        num_sample_points = 1000
```

Comment:
That is a lot of parameters here. I would create a private const/static struct at the top of the script so it is easily accessible, or capitalize these so we know they are hard coded unchangeable magic numbers. But this is a question of **preference**. Feel free to ignore.